mlflow/promptflow/__init__.py

```diff
@@ -0,0 +1,393 @@
+    metadata=None):
+    """
+    Save a Promptflow model to a path on the local file system.
+
+    :param model: A promptflow model loaded by `promptflow.load_flow()`.
+    :param path: Local path where the serialized model (as YAML) is to be saved.
+    :param conda_env: {{ conda_env }}
+    :param code_paths: A list of local filesystem paths to Python file dependencies (or directories
+                       containing file dependencies). These files are *prepended* to the system
+                       path when the model is loaded.
+    :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+    :param signature: :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+                      describes model input and output
+                      :py:class:`Schema <mlflow.types.Schema>`.
+                      If not specified, the model signature would be set according to
+                      `model.inputs` and `model.outputs` as columns names, `DataType.int`,
+                      `DataType.double`and `DataType.boolean` as the column type for
+                      promptflow type `int`, `double` and `bool`, and `DataType.string`
+                      as the column type for the other promptflow types.
+
+    :param input_example: {{ input_example }}
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+    """
+    import promptflow
+    from promptflow import load_flow
+    from promptflow.contracts.tool import ValueType
+    from promptflow._sdk.entities._flow import Flow
+    from promptflow._sdk._utils import _merge_local_code_and_additional_includes
+    from promptflow._sdk.operations._run_submitter import remove_additional_includes
+
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+    # check if path exists
+    path = os.path.abspath(path)
+    _validate_and_prepare_target_save_path(path)
+
+    code_dir_subpath = _validate_and_copy_code_paths(code_paths, path)
+
+    model_flow_path = os.path.join(path, _MODEL_FLOW_DIRECTORY)
+
+    if not isinstance(model, Flow):
+        raise mlflow.MlflowException.invalid_parameter_value(
+            _UNSUPPORTED_MODEL_ERROR_MESSAGE.format(instance_type=type(model).__name__)
+        )
+
+    # Resolve additional includes in flow
+    with _merge_local_code_and_additional_includes(code_path=model.code) as resolved_model_dir:
+        remove_additional_includes(Path(resolved_model_dir))
+        shutil.copytree(src=resolved_model_dir, dst=model_flow_path)
+    # Get flow env in flow dag
+    flow_env = _resolve_env_from_flow(model.flow_dag_path)
+
+    def _parse_pf_type(type_str):
```

Comment:
minor point: if you instead put a .to_mlflow_signature() in the pf model object directly, iterating on this code would be easier, but that's not a big deal

Reply:
Thanks, I prefer leaving them here, because promptflow package doesn't include mlflow in dependencies for now.